src/dotnet-svcutil/lib/src/Shared/Options/UpdateOptions.cs

```diff
@@ -97,6 +97,16 @@ public void MakePathsRelativeTo(DirectoryInfo optionsFileDirectory)
                     this.Inputs[idx] = new Uri(relPath, UriKind.Relative);
                 }
             }
+
+            //update referenced assembly path
+            for (int idx = 0; idx < this.References.Count; idx++)
+            {
+                var reference = this.References[idx];
+                if (reference.DependencyType.Equals(ProjectDependencyType.Binary))
+                {
+                    this.References[idx] = ProjectDependency.FromAssembly(reference.AssemblyName);
```

Comment:
Only binary dependency type is affected, reconstruct the reference with reference.AssemblyName ensures name instead of path get written to the .json file.